pkg/collectors/service.go

```diff
@@ -110,5 +138,33 @@ func generateServiceMetrics(obj interface{}) []*metrics.Metric {
 	labelKeys, labelValues := kubeLabelsToPrometheusLabels(s.Labels)
 	addGauge(serviceLabelsDesc(labelKeys), 1, labelValues...)
 
+	if s.Spec.ExternalName != "" {
+		addGauge(descServiceExternalName, 1, s.Spec.ExternalName)
+	} else {
+		addGauge(descServiceExternalName, 0, s.Spec.ExternalName)
+	}
+
+	if s.Spec.LoadBalancerIP != "" {
+		addGauge(descServiceLoadBalancerIP, 1, s.Spec.LoadBalancerIP)
+	} else {
+		addGauge(descServiceLoadBalancerIP, 0, s.Spec.LoadBalancerIP)
+	}
+
+	if len(s.Status.LoadBalancer.Ingress) > 0 {
+		for _, ingress := range s.Status.LoadBalancer.Ingress {
+			addGauge(descServiceStatusLoadBalancerIngress, 1, ingress.IP, ingress.Hostname)
+		}
+	} else {
+		addGauge(descServiceStatusLoadBalancerIngress, 0, "", "") // Add empty for completeness
+	}
+
+	if len(s.Spec.ExternalIPs) > 0 {
+		for _, external_ip := range s.Spec.ExternalIPs {
+			addGauge(descServiceSpecExternalIP, 1, external_ip)
+		}
+	} else {
+		addGauge(descServiceSpecExternalIP, 0, "") // Add empty for completeness
```

Comment:
I don't think we should do these, if there is no external IP/loadbalancer ingress IP then there should be no metric